ever_given/ever_given/engines/docker_engine.py

```diff
@@ -77,3 +78,23 @@ def run_container(
             detach=True,
         )
         return DockerContainerInstance(client, docker_container)
+
+    @classmethod
+    def pull_container(cls, container_uri, container_type, save_path=None, aws_login_func=None):
+        # pylint: disable=unused-argument
+        if container_type not in DOCKER_CONTAINER_TYPES:
+            raise ValueError(f"Container type: {container_type} not supported by Docker Engine")
+
+        if aws_login_func:
+            aws_login_func("singularity")
+
+        pull_cmd = ['docker', 'pull', container_uri]
+
+        ended_proc = subprocess.run(
+            pull_cmd, capture_output=True
```

Comment:
```suggestion
            pull_cmd, capture_output=True, check=True
```
to throw an error on failure